File: storage/db-core/src/main/java/com/coffee/gu/menu/OptionEntity.java
package com.coffee.gu.menu;

import com.coffee.gu.BaseEntity;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;

import java.math.BigDecimal;

@Table(name = "option")
@Entity
public class OptionEntity extends BaseEntity {

    private Long optionGroupId;
    private String name;
    private BigDecimal extraPrice;

    public OptionEntity() {}

    public OptionEntity(Long optionGroupId, String name, BigDecimal extraPrice) {
        this.optionGroupId = optionGroupId;
        this.name = name;
        this.extraPrice = extraPrice;
    }

    public Option toModel() {
        return new Option(id, optionGroupId, name, extraPrice);
    }

    public Long getOptionGroupId() {
        return optionGroupId;
    }

    public String getName() {
        return name;
    }

    public BigDecimal getExtraPrice() {
        return extraPrice;
    }
}

File: storage/db-core/src/main/kotlin/com/coffee/gu/menu/OptionEntity.kt
package com.coffee.gu.menu

import com.coffee.gu.BaseEntity
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.math.BigDecimal

@Table(name = "option")
@Entity
class OptionEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0,
    val optionGroupId: Long,
    val name: String,
    val extraPrice: BigDecimal,
) : BaseEntity() {

    fun toModel(): Option {
        return Option(
            id = id,
            optionGroupId = optionGroupId,
            name = name,
            extraPrice = extraPrice,
        )
    }
}
